ipydatagrid/datagrid.py

```diff
@@ -70,7 +70,7 @@ def all(self):
         represented as a dictionary
         with keys 'r': row and 'c': column
         """
-        return list(self)
+        return [cell for cell in self]  # noqa: C416
```

Comment:
Why do we need this change? flake8 also agrees we should not, hence the override you've added, but it would be good to know why

Reply:
list(self) returns an empty list, whereas the list comprehension returns the correct output. I didn't look into why, specifically, as I was more concerned with getting this to work first. If we need to refactor the class, I can do it separately

Reply:
IMHO we should look into it why. Maybe the actual bug is not here but somewhere else, and here we're applying just a symptom fix rather than the proper solution.

Reply:
> As it involves UI manipulation to trigger the update, there is no real way of testing it purely on the Python side. We are in the final stages of open sourcing Galata which has visual regression end-to-end testing exactly for these situations.

Sure testing the symptom is UI manipulation testing. So we might not be able to do that. However, you've said:

> list(self) returns an empty list, whereas the list comprehension returns the correct output.

which sounds like a pure Python problem, and this class is a pure Python class, so perhaps testing the issue of ``list(self) returns an empty list`` part can be tested.

Reply:
I can't find anything incriminating there. Can you take a quick look at the SelectionHelper class and let me know if anything stands out for you?

Reply:
> > As it involves UI manipulation to trigger the update, there is no real way of testing it purely on the Python side. We are in the final stages of open-sourcing Galata which has visual regression end-to-end testing exactly for these situations.
> 
> Sure testing the symptom is UI manipulation testing. So we might not be able to do that, just yet. However, you've said:
> 
> > list(self) returns an empty list, whereas the list comprehension returns the correct output.
> 
> which sounds like a pure Python problem, and this class is a pure Python class, so perhaps testing the issue of `list(self) returns an empty list` part can be tested.

The correct output in this case is a list of cells which have been selected by the user from the rendered grid. I suppose we can use the selection API and call it from python directly.

Reply:
So found the underlying issue, added a commit that fixes it. The test you've added never hit the code section in question, so I amended that and used a parameterization to avoid code duplication in the test suite.